src/applications/edu-benefits/components/AutosuggestField.jsx

```diff
@@ -0,0 +1,296 @@
+      });
+    }
+  }
+
+  getSuggestions = (options, value) => {
+    if (value) {
+      const uiOptions = this.props.uiSchema['ui:options'];
+      const sortedOptions = sortListByFuzzyMatch(value, options).slice(0, uiOptions.maxOptions);
+      return sortedOptions;
+    }
+
+    return options;
+  }
+
+  getFormData = (suggestion) => {
+    if (this.useEnum) {
+      return suggestion.id;
+    }
+
+    // When freeInput is true, we'll return the label to the api instead of the id
+    if (this.props.uiSchema['ui:options'].freeInput) {
+      return suggestion.label;
+    }
+
+    return _.set('widget', 'autosuggest', suggestion);
+  }
+
+  handleClearSelection = (e, downshiftClear) => {
+    e.preventDefault();
+    // using Downshift's clear sets the correct focus state after using clear
+    downshiftClear(() => this.setState({
+      input: '',
+      searchTerm: '',
+      loading: false
+    }));
+  }
+
+  handleInputValueChange = (inputValue) => {
+    if (inputValue !== this.state.input) {
+      const uiOptions = this.props.uiSchema['ui:options'];
+      if (uiOptions.queryForResults) {
+        this.setState({
+          loading: true,
+          searchTerm: inputValue
+        });
+        this.debouncedGetOptions(inputValue);
+      }
+
+      let item = { widget: 'autosuggest', label: inputValue };
+      // once the input is long enough, check for exactly matching strings so that we don't
+      // force a user to click on an item when they've typed an exact match of a label
+      if (inputValue && inputValue.length > 3) {
+        const matchingItem = this.state.suggestions.find(suggestion => suggestion.label === inputValue);
+        if (matchingItem) {
+          item = this.getFormData(matchingItem);
+        }
+      }
+
+      this.props.onChange((this.props.uiSchema['ui:options'].freeInput || this.useEnum) ? inputValue : item);
+      this.setState({
+        input: inputValue,
+        suggestions: this.getSuggestions(this.state.options, inputValue)
+      });
+    } else if (inputValue === '') {
+      this.props.onChange();
+      this.setState({
+        input: inputValue,
+        suggestions: this.getSuggestions(this.state.options, inputValue)
+      });
+    }
+  }
+
+  handleChange = (selectedItem) => {
+    const value = this.getFormData(selectedItem);
+    this.props.onChange(value);
+    if (this.state.input !== selectedItem.label) {
+      this.setState({
+        input: selectedItem.label,
+      });
+    }
+  }
+
+  handleKeyDown = (event) => {
+    if (event.keyCode === ESCAPE_KEY) {
+      this.setState({ input: '' });
+    }
+  }
+
+  handleBlur = () => {
+    this.props.onBlur(this.props.idSchema.$id);
+  }
+
+  render() {
+    const { formContext, formData, uiSchema, schema } = this.props;
+
+    const {
+      loading,
+      options: items
+    } = this.state;
+    const listLabel = loading ? 'Loading...' : this.state.listLabel;
+
+    if (formContext.reviewMode) {
```

Comment:
review and edit mode are pretty broken-  this `z-index` might cause some issues. 
![screen shot 2018-07-19 at 09 08 53](https://user-images.githubusercontent.com/4998130/42951423-60a86f84-8b33-11e8-9fa7-8323787e44d0.png)
![screen shot 2018-07-19 at 09 10 12](https://user-images.githubusercontent.com/4998130/42951513-928c8daa-8b33-11e8-810e-f780634f363b.png)